app/models/piece.rb

```diff
@@ -32,4 +32,24 @@ def set_image
     color ? color_string = "white" : color_string = "black"
     self.image_name ||= "#{color_string}-#{piece_type.downcase}.png"
   end
+
+  def move_to!(new_x, new_y)
```

Comment:
@nicolaguidi @acodeinprogress we will need to determine how we want to implement the actual moving of a piece. the valid_move? method (which itself should call is_obstructed?) for each type of piece needs to be called before this move_to! function. we could implement this by having the update action in the piece controller always call valid_move?  on the piece in quest before using this move_to! function, but i'm not sure i like that (since move_to! currently always depends on the programmer to call valid_move? beforehand....i think it would be safer/cleaner to call valid_move? from inside move_to! so it's always called no matter what) that's just my take on it though, let me know what you guys think, @acodeinprogress will be implementing the first go at the piece controller update action.